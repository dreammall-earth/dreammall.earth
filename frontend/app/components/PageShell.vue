<template>
  <v-app>
    <slot />
  </v-app>
</template>
