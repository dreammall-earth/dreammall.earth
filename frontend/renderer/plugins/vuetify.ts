// eslint-disable-next-line import/no-unassigned-import
import '@mdi/font/css/materialdesignicons.css'
// eslint-disable-next-line import/no-unassigned-import
import 'vuetify/styles'
import { I18n, useI18n } from 'vue-i18n'
import { ThemeDefinition, createVuetify } from 'vuetify'
import { aliases as mdiAliases } from 'vuetify/iconsets/mdi'
import { VNumberInput } from 'vuetify/labs/VNumberInput'
import { createVueI18nAdapter } from 'vuetify/locale/adapters/vue-i18n'

import { aliases as customAliases } from '#assets/icons'
import dark from '#assets/scss/dark.module.scss'
import light from '#assets/scss/light.module.scss'

function makeThemeFromCssModule(theme: CSSModuleClasses, isDark: boolean): ThemeDefinition {
  return {
    dark: isDark,
    colors: {
      background: theme.backgroundColor,
      surface: theme.surfaceColor,
      primary: theme.primaryColor,
      secondary: theme.secondaryColor,
      success: theme.successColor,
      info: theme.infoColor,
      warning: theme.warningColor,
      error: theme.errorColor,
      font: theme.fontColor,
      'dropdown-background': theme.dropdownBackgroundColor,
      'cockpit-highlight': theme.cockpitHighlightColor,
      'dm-panel-submit-color': theme.dmPanelSubmitColor,
      'dm-panel-submit-background-color': theme.dmPanelSubmitBackgroundColor,
      'dm-panel-text-input-color': theme.dmPanelTextInputColor,
      'dm-panel-call-action-button-color': theme.dmPanelCallActionButtonColor,
      'dm-panel-call-action-button-background-color': theme.dmPanelCallActionButtonBackgroundColor,
      'dm-panel-call-action-button-indicator-color': theme.dmPanelCallActionButtonIndicatorColor,
      'dm-panel-call-action-button-indicator-background-color':
        theme.dmPanelCallActionButtonIndicatorBackgroundColor,
      'dm-panel-reminder-text-color': theme.dmPanelReminderTextColor,
      'dm-panel-reminder-link-color': theme.dmPanelReminderLinkColor,
      'dm-switch-inactive-color': theme.dmSwitchInactiveColor,
      'dm-switch-active-color': theme.dmSwitchActiveColor,
    },
    variables: {
      'border-color': theme.borderColor,
      'border-opacity': theme.borderOpacity,
      // Vuetify colors don't support opacity, so we use variables instead.
      // Beware that HEX values won't work.
      'icon-background': theme.iconBackground,
      'bottom-menu-background': theme.bottomMenuBackground,
      'sidebar-background': theme.sidebarBackground,
      'cockpit-input-background': theme.cockpitInputBackground,
      'cockpit-element-background': theme.cockpitElementBackground,
      'cockpit-chip-background': theme.cockpitChipBackground,
      'cockpit-chip-background-2': theme.cockpitChipBackground2,
      'panel-background': theme.panelBackground,
      'panel-element-background': theme.panelElementBackground,
      'dm-panel-text-input-background-color': theme.dmPanelTextInputBackgroundColor,
      'dm-panel-background-color': theme.dmPanelBackgroundColor,
      'dm-panel-border-color': theme.dmPanelBorderColor,
      'dm-panel-reminder-text-background-color': theme.dmPanelReminderTextBackgroundColor,
      'dm-panel-reminder-link-background-color': theme.dmPanelReminderLinkBackgroundColor,
    },
  }
}

const themes = {
  light: makeThemeFromCssModule(light, false),
  dark: makeThemeFromCssModule(dark, true),
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default (i18n: I18n<any, NonNullable<unknown>, NonNullable<unknown>, string, false>) =>
  createVuetify({
    components: {
      VNumberInput,
    },
    locale: {
      adapter: createVueI18nAdapter({ i18n, useI18n }),
    },
    ssr: true,
    icons: {
      aliases: { ...mdiAliases, ...customAliases },
    },
    theme: {
      defaultTheme: 'light',
      themes,
    },
  })
