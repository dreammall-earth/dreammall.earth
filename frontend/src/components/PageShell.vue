<template>
  <v-app>
    <slot />
  </v-app>
</template>

<script lang="ts" setup>
import TopMenu from '#components/menu/TopMenu.vue'
</script>
