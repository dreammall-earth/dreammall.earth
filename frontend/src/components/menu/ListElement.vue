<template>
  <v-card class="mx-auto" width="400">
    <v-list lines="two">
      <v-list-item
        v-for="(item, index) in items"
        :key="index"        
        :class="{ 'full-width': item.fullWidth, rounded: item.rounded, 'custom-list-item': true }"
        @click="handleItemClick"
        :title="item.title"
        :subtitle="item.subtitle"
      >
        <template v-if="item.prepend" #prepend>
          <component :is="item.prepend" v-bind="item.prependProps" />
        </template>

        <!-- <v-list-item-title>{{ item.title }}</v-list-item-title>
        <v-list-item-subtitle v-if="item.subtitle">{{ item.subtitle }}</v-list-item-subtitle> -->

        <template v-if="item.append" #append>
          <component :is="item.append" v-bind="item.appendProps" />
        </template>
      </v-list-item>
    </v-list>
  </v-card>
</template>

<script lang="ts" setup>
import { PropType } from 'vue'

interface Item {
  title: string
  subtitle?: string
  fullWidth: boolean
  rounded?: boolean
  prepend?: string | object
  prependProps?: object
  append?: string | object
  appendProps?: object
}

defineProps({
  items: {
    type: Array as PropType<Item[]>,
    required: true,
  },
})
const emit = defineEmits(['item-click'])

const handleItemClick = () => {
  closeMenu()
  emit('item-click')
}

const closeMenu = () => {
  // Logik zum Schließen des Menüs hinzufügen, falls erforderlich
}
</script>

<style scoped>
.custom-list-item {
  margin-bottom: 8px;
  padding: 8px 16px;
  border-radius: 8px;
  background-color: #ffffff;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.full-width {
  width: 100%;
}

.rounded {
  border-radius: 16px;
}

.v-list-item-content {
  display: flex;
  align-items: center;
  flex-grow: 1;
}

.v-list-item-title {
  font-weight: 500;
  font-size: 16px;
  margin-right: 8px;
}

.v-list-item-subtitle {
  font-size: 14px;
  color: gray;
}
</style>
