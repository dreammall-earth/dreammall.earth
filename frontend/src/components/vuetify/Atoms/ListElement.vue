<template>
  <v-card class="mx-auto" width="400">
    <v-list lines="two">
      <v-list-item
        v-for="(item, index) in items"
        :key="index"
        :class="{ 'custom-list-item': true }"
        :title="item.meetingName"
        :subtitle="$t('rooms.participantCount', { count: item.participantCount })"
        @click="handleItemClick(item.joinLink)"
      >
        <!-- 
        <template v-if="item.prepend" #prepend>
          <component :is="item.prepend" v-bind="item.prependProps" />
        </template>
        -->

        <!-- <v-list-item-title>{{ item.title }}</v-list-item-title>
             <v-list-item-subtitle v-if="item.subtitle">{{ item.subtitle }}</v-list-item-subtitle> -->
        <!-- 
        <template v-if="item.append" #append>
          <component :is="item.append" v-bind="item.appendProps" />
        </template>
        -->
      </v-list-item>
    </v-list>
  </v-card>
</template>

<script lang="ts" setup>
import { navigate } from 'vike/client/router'
import { PropType } from 'vue'

import { useActiveRoomStore } from '#stores/activeRoomStore'
import { Room } from '#stores/roomsStore'

const activeRoomStore = useActiveRoomStore()

/*
    export interface Item {
    title: string
    subtitle?: string
    rounded?: boolean
    prepend?: string | object
    prependProps?: object
    append?: string | object
    appendProps?: object
    }
  */

defineProps({
  items: {
    type: Array as PropType<Room[]>,
    required: true,
  },
})

// const emit = defineEmits(['item-click'])

const handleItemClick = (link: string) => {
  console.log('handleItemClick', link)
  closeMenu()
  activeRoomStore.setActiveRoom(link)
  navigate('/room/')
  // emit('item-click')
}
const closeMenu = () => {
  // Logik zum Schließen des Menüs hinzufügen, falls erforderlich
}
</script>

<style scoped>
.custom-list-item {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 8px 16px;
  margin-bottom: 8px;
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0 2px 4px rgb(0 0 0 / 10%);
}
</style>
