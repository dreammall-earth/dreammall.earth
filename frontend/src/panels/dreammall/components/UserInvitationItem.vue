<template>
  <v-list-item class="user-invitation-item my-2 pa-2 pr-3 rounded-pill" elevation="0" rounded>
    <template #prepend>
      <v-avatar size="40">
        <v-img v-if="user.avatar" :src="user.avatar" :alt="user.name" cover></v-img>
        <svg v-else width="40" height="40" viewBox="0 0 40 40" xmlns="http://www.w3.org/2000/svg">
          <circle cx="20" cy="20" r="20" fill="#E0E0E0" />
          <circle cx="20" cy="15" r="7" fill="#BDBDBD" />
          <path
            d="M6 34.5C6 27.5964 11.5964 22 18.5 22H21.5C28.4036 22 34 27.5964 34 34.5V40H6V34.5Z"
            fill="#BDBDBD"
          />
        </svg>
      </v-avatar>
    </template>

    <v-list-item-title class="text-white font-weight-medium">
      {{ user.name }}
    </v-list-item-title>

    <template #append>
      <v-btn
        elevation="0"
        :color="user.invited ? 'success' : 'secondary'"
        class="invitation-button text-caption py-2 px-3 rounded-pill"
        size="small"
        @click="toggleInvitation"
      >
        {{
          user.invited
            ? $t('dream-mall-panel.setup.invitation-sent')
            : $t('dream-mall-panel.setup.invite')
        }}
      </v-btn>
    </template>
  </v-list-item>
</template>

<script lang="ts" setup>
import { defineProps, defineEmits } from 'vue'

import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const props = defineProps<{
  user: UserInvitation
}>()

const emit = defineEmits<{
  (e: 'update:invited', value: boolean): void
}>()

const toggleInvitation = () => {
  emit('update:invited', !props.user.invited)
}
</script>

<style lang="scss" scoped>
.user-invitation-item {
  --background-user-item: #5d6670; // todo: save globally
  --background-invitation-button: #8b949b; // todo: save globally

  background-color: var(--background-user-item) !important;

  .invitation-button {
    &.bg-secondary {
      background-color: var(--background-invitation-button) !important;
    }
  }
}
</style>
