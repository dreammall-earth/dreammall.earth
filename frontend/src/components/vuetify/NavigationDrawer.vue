<template>
  <v-navigation-drawer v-model="drawer" :location="computedLocation" temporary width="400">
    <slot />
  </v-navigation-drawer>
</template>

<script lang="ts" setup>
import { computed, ref, watch } from 'vue'
import { useDisplay } from 'vuetify'

export type DrawerLocation = 'right' | 'bottom' | 'left' | 'end' | 'top' | 'start'

const props = withDefaults(
  defineProps<{
    modelValue: boolean
    location?: DrawerLocation
  }>(),
  {
    location: 'right',
  },
)

const emits = defineEmits(['update:modelValue'])
const display = useDisplay()
const drawer = ref(props.modelValue)

const computedLocation = computed(() => {
  return display.mobile.value ? 'bottom' : props.location
})

watch(
  () => props.modelValue,
  (newVal) => {
    drawer.value = newVal
  },
)

watch(
  () => drawer.value,
  (newVal) => {
    if (newVal !== props.modelValue) {
      emits('update:modelValue', newVal)
    }
  },
)
</script>
