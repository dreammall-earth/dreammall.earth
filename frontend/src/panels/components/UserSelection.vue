<template>
  <v-text-field
    v-model="userSearch"
    rounded
    clearable
    class="elevation-0 w-100"
    content-class="elevation-0"
    label="Suche"
    variant="solo-filled"
    append-inner-icon="mdi-magnify"
  />

  <div class="user-list-container w-100">
    <div v-if="error" class="text-center text-error">
      {{ $t('dream-mall-panel.setup.search-error') }}
    </div>

    <v-list
      v-if="displayedUsers.length"
      class="bg-transparent w-100 user-list"
      :class="{ loading: isLoading }"
    >
      <UserInvitationItem
        v-for="user in displayedUsers"
        :key="user.id"
        :user="user"
        @update:invited="updateInvitationStatus(user.id, $event)"
      />
    </v-list>

    <div v-if="!displayedUsers.length" class="text-center mt-4 text-grey-darken-2">
      {{ $t('dream-mall-panel.setup.no-result') }}
    </div>
  </div>
</template>

<script lang="ts" setup>
import { ref, computed, watch } from 'vue'

import UserInvitationItem from '#src/panels/components/UserInvitationItem.vue'
import useSearchUsers from '#src/panels/composables/useSearchUsers'
import UserInvitation from '#src/panels/dreammall/interfaces/UserInvitation'

const users = defineModel<number[]>({ required: true })

const { searchUsers, searchResults, isLoading, error } = useSearchUsers()
const userSearch = ref('')

watch(userSearch, (newSearch) => {
  searchUsers(newSearch)
})

const displayedUsers = computed<UserInvitation[]>(() => {
  return searchResults.value.map((user) => ({
    id: user.id,
    name: user.name || user.username,
    avatar: null, // not supporting yet
    invited: users.value.includes(user.id), // todo: user already included in myTable?
  }))
})

const updateInvitationStatus = (userId: number, invited: boolean) => {
  let currentUsers: number[] = [...users.value]
  if (invited && !currentUsers.includes(userId)) {
    currentUsers.push(userId)
  } else if (!invited && currentUsers.includes(userId)) {
    currentUsers = currentUsers.filter((id) => id !== userId)
  }
  users.value = currentUsers
}
</script>

<style lang="scss" scoped>
.flat-text-field .v-field--variant-solo-filled {
  box-shadow: none !important;
}

.user-list-container {
  height: 200px;
  overflow-y: auto;

  .user-list {
    height: 100%;

    &.loading {
      opacity: 0.5;
      transition: opacity 0.5s ease;
    }
  }
}
</style>
