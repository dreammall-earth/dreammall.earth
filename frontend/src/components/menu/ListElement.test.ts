import { mount, VueWrapper } from '@vue/test-utils'
import { describe, it, expect, beforeEach, vi } from 'vitest'
import { createVuetify } from 'vuetify'
import ListElement from './ListElement.vue'
import { VIcon, VBtn } from 'vuetify/components'

interface Item {
  title: string
  subtitle?: string
  fullWidth: boolean
  rounded?: boolean
  prepend?: string | object
  prependProps?: object
  append?: string | object
  appendProps?: object
}

describe('ListElement', () => {
  let wrapper: VueWrapper<InstanceType<typeof ListElement>>
  const vuetify = createVuetify()

  beforeEach(() => {
    wrapper = mount(ListElement, {
      global: {
        plugins: [vuetify],
      },
      props: {
        items: [
          { title: 'Item 1', fullWidth: false, append: VBtn, appendProps: { icon: 'mdi-menu' } },
          { title: 'Item 2', fullWidth: true, append: VIcon, appendProps: { icon: 'mdi-menu' } },
        ] as Item[],
      },
    })
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders list items', () => {
    expect(wrapper.findAll('.custom-list-item').length).toBe(2)
  })

  it('renders append slot correctly', () => {
    const appendIcons = wrapper.findAllComponents(VIcon)
    const appendButtons = wrapper.findAllComponents(VBtn)
    expect(appendIcons.length + appendButtons.length).toBe(2)
  })

  it('handles item click and closes menu', async () => {
    const emitSpy = vi.spyOn(wrapper.vm, '$emit')
    await wrapper.findAll('.custom-list-item')[0].trigger('click')
    expect(emitSpy).toHaveBeenCalledWith('item-click')
    // Sie können hier zusätzliche Tests hinzufügen, um zu prüfen, ob das Menü geschlossen wird.
  })
})
