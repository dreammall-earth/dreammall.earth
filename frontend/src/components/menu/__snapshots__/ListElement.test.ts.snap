// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListElement > renders correctly 1`] = `
<div
  class="v-card v-theme--light v-card--density-default v-card--variant-elevated mx-auto"
  data-v-e7d277f5=""
  style="width: 400px;"
>
  <!---->
  <div
    class="v-card__loader"
  >
    <div
      aria-hidden="true"
      aria-valuemax="100"
      aria-valuemin="0"
      class="v-progress-linear v-theme--light v-locale--is-ltr"
      role="progressbar"
      style="top: 0px; height: 0px; --v-progress-linear-height: 2px;"
    >
      <!---->
      <div
        class="v-progress-linear__background"
        style="width: 100%;"
      />
      <transition-stub
        appear="false"
        css="true"
        name="fade-transition"
        persisted="false"
      >
        <div
          class="v-progress-linear__indeterminate"
        >
          
          <div
            class="v-progress-linear__indeterminate long"
          />
          <div
            class="v-progress-linear__indeterminate short"
          />
          
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  
  <div
    class="v-list v-theme--light v-list--density-compact v-list--one-line"
    data-v-e7d277f5=""
    role="listbox"
    tabindex="0"
  >
    
    
    <div
      class="v-list-item v-list-item--link v-theme--light v-list-item--density-compact v-list-item--one-line rounded-0 v-list-item--variant-plain custom-list-item"
      data-v-e7d277f5=""
      tabindex="-2"
    >
      
      <span
        class="v-list-item__overlay"
      />
      <span
        class="v-list-item__underlay"
      />
      
      <!---->
      <div
        class="v-list-item__content"
        data-no-activator=""
      >
        <!---->
        <!---->
        
        <div
          class="v-list-item-title"
          data-v-e7d277f5=""
        >
          Item 1
        </div>
        <!--v-if-->
        
      </div>
      <div
        class="v-list-item__append"
      >
        
        
        <button
          class="v-btn v-btn--elevated v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-elevated"
          data-v-e7d277f5=""
          type="button"
        >
          
          <span
            class="v-btn__overlay"
          />
          <span
            class="v-btn__underlay"
          />
          
          <!---->
          <span
            class="v-btn__content"
            data-no-activator=""
          >
            <i
              aria-hidden="true"
              class="mdi-menu mdi v-icon notranslate v-theme--light v-icon--size-default"
            />
          </span>
          <!---->
          <!---->
        </button>
        
        
        <div
          class="v-list-item__spacer"
        />
      </div>
    </div>
    <div
      class="v-list-item v-list-item--link v-theme--light v-list-item--density-compact v-list-item--one-line rounded-0 v-list-item--variant-plain custom-list-item full-width"
      data-v-e7d277f5=""
      tabindex="-2"
    >
      
      <span
        class="v-list-item__overlay"
      />
      <span
        class="v-list-item__underlay"
      />
      
      <!---->
      <div
        class="v-list-item__content"
        data-no-activator=""
      >
        <!---->
        <!---->
        
        <div
          class="v-list-item-title"
          data-v-e7d277f5=""
        >
          Item 2
        </div>
        <!--v-if-->
        
      </div>
      <div
        class="v-list-item__append"
      >
        
        
        <i
          aria-hidden="true"
          class="mdi-menu mdi v-icon notranslate v-theme--light v-icon--size-default"
          data-v-e7d277f5=""
        />
        
        
        <div
          class="v-list-item__spacer"
        />
      </div>
    </div>
    
    
  </div>
  
  <!---->
  
  <!---->
  <span
    class="v-card__underlay"
  />
  
</div>
`;
